### PR TITLE
mtx3: add getHalfHourDemandExport, getHalfHourDemandVareExport, getHalfHourDemandVariExport commands

### PR DESCRIPTION
**Pull request by aleks-and**

### CODE REVIEW

**Review by DeadByElpy — APPROVED** [review]

File: src/mtx/commands/downlink/getHalfHourDemandExport.ts [hunk 1]
```diff
@@ -1,6 +1,8 @@
 /**
  * Downlink command to get active energy (`A-`) in half hours by date.
```
Comment (DarkPark) [inline src/mtx/commands/downlink/getHalfHourDemandExport.ts:2]:
```suggestion
 * Downlink command to get active energy (`A-`) in half hours by date for MTX1 and ...
```

File: src/mtx3/commands/uplink/getHalfHourDemandExport.ts [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+
+
+/**
+ * Encode command parameters.
+ *
+ * @param parameters - command payload
+ * @returns full message (header with body)
+ */
+export const toBytes = ( parameters: IGetHalfHourDemandResponseParameters ): types.TBytes => {
+    const buffer: ICommandBinaryBuffer = new CommandBinaryBuffer(parameters.energies.length > MIN_HALF_HOUR_PERIODS ? MAX_HALF_HOUR_COMMAND_SIZE : MIN_HALF_HOUR_COMMAND_SIZE);
```
Comment (DarkPark) [inline src/mtx3/commands/uplink/getHalfHourDemandExport.ts:161]: too long line

File: src/mtx/commands/downlink/getHalfHourDemandExport.ts (hunk 1 above)
Comment (aleks-and) [reply src/mtx/commands/downlink/getHalfHourDemandExport.ts:2]: fixed

File: src/mtx3/commands/uplink/getHalfHourDemandExport.ts (hunk 2 above)
Comment (aleks-and) [reply src/mtx3/commands/uplink/getHalfHourDemandExport.ts:161]: fixed here and in other similar places

**Review by DarkPark — APPROVED** [review]

**Pull request merged**